### ISSUE TITLE
[Question]: Support for Clair3 model r1041_e82_400bps_sup_g615 using R10.4.1 flowcell

### ISSUE THREAD

**Issue by gilmansiugroup** [CFIA-NCFAD/nf-flu#11]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Bug/Issue

Hi @peterk87,

We would like to ask whether the `r1041_e82_400bps_sup_g615` model is compatible with the `--clair3_variant_model` option? Thank you very much!

Best regards,
Eddie

### Nextflow command-line

```shell
-
```

### Error Message

```shell
-
```

### Workflow Version

v3.1.3

### Nextflow Version

21.10.4.5656

### Hardware

Desktop

### Conda/Container Engine

Docker

**Comment by peterk87** [comment]
Addressed this issue in #12 and release [3.1.4](https://github.com/CFIA-NCFAD/nf-flu/releases/tag/3.1.4)

> This release addresses issue [#11](https://github.com/CFIA-NCFAD/nf-flu/issues/11) adding a new option `--clair3_user_variant_model <PATH TO CLAIR3 MODEL>` to allow user to provide a Clair3 model not included with Clair3, e.g. a [Rerio](https://github.com/nanoporetech/rerio) Clair3 model for r10 flowcells.

@gilmansiugroup please let us know if that addresses your issue if you get a chance. Thanks for reporting the issue!

**Comment by llk578496** [comment]
Hi @peterk87 @nhhaidee ,

We just downloaded the `r1041_e82_400bps_sup_g615` model files from Rerio with the following structure:
```
tree /mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615
/mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615
├── combine_models.log
├── full_alignment.data-00000-of-00002
├── full_alignment.data-00001-of-00002
├── full_alignment.index
├── pileup.data-00000-of-00002
├── pileup.data-00001-of-00002
└── pileup.index

0 directories, 7 files
```
Then, we tried to run with the following command:
`nextflow run CFIA-NCFAD/nf-flu --input /mnt/data6/FLU/04_main/04.1_tmp/23MB310073/nf-flu/samplesheet.csv --platform nanopore -profile docker --outdir /mnt/data6/FLU/04_main/04.1_tmp/23MB310073/nf-flu/results --clair3_user_variant_model /mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615 --low_coverage 20`

However, we encountered the following error in the `process > NF_FLU:NANOPORE:CLAIR3` with the following error log:

```
executor >  local (34)
[7d/68d546] process > NF_FLU:NANOPORE:CHECK_SAMPLE_SHEET (1)                      [100%] 1 of 1 ✔
[-        ] process > NF_FLU:NANOPORE:READ_COUNT_FAIL_TSV                         -
[41/c0b2bb] process > NF_FLU:NANOPORE:READ_COUNT_PASS_TSV                         [100%] 1 of 1 ✔
[15/6cf29d] process > NF_FLU:NANOPORE:GUNZIP_NCBI_FLU_FASTA (influenza.fna.gz)    [100%] 1 of 1 ✔
[fb/9d4fdc] process > NF_FLU:NANOPORE:BLAST_MAKEBLASTDB_NCBI (influenza.fna)      [100%] 1 of 1 ✔
[a7/ecef03] process > NF_FLU:NANOPORE:CAT_NANOPORE_FASTQ (23MB310073)             [100%] 1 of 1 ✔
[a4/ec8b15] process > NF_FLU:NANOPORE:IRMA (23MB310073)                           [100%] 1 of 1 ✔
[55/f293dc] process > NF_FLU:NANOPORE:BLAST_BLASTN_IRMA (23MB310073)              [100%] 1 of 1 ✔
[a8/e5952c] process > NF_FLU:NANOPORE:PULL_TOP_REF_ID (23MB310073)                [100%] 1 of 1 ✔
[f8/fcf67c] process > NF_FLU:NANOPORE:SEQTK_SEQ (23MB310073|3_PA|MT499741)        [100%] 8 of 8 ✔
[d6/61f518] process > NF_FLU:NANOPORE:MINIMAP2 (23MB310073|3_PA|MT499741)         [100%] 6 of 6
[14/ce73b6] process > NF_FLU:NANOPORE:MOSDEPTH_GENOME (23MB310073|2_PB1|MN975855) [ 50%] 1 of 2
[36/b13407] process > NF_FLU:NANOPORE:CLAIR3 (23MB310073|2_PB1|MN975855)          [ 50%] 1 of 2, failed: 1
[-        ] process > NF_FLU:NANOPORE:BCF_FILTER_CLAIR3                           -
[-        ] process > NF_FLU:NANOPORE:VCF_FILTER_FRAMESHIFT                       -
[-        ] process > NF_FLU:NANOPORE:BCFTOOLS_STATS                              -
[-        ] process > NF_FLU:NANOPORE:COVERAGE_PLOT                               -
[-        ] process > NF_FLU:NANOPORE:BCF_CONSENSUS                               -
[-        ] process > NF_FLU:NANOPORE:CAT_CONSENSUS                               -
[-        ] process > NF_FLU:NANOPORE:BLAST_BLASTN_CONSENSUS                      -
[-        ] process > NF_FLU:NANOPORE:SUBTYPING_REPORT_BCF_CONSENSUS              -
[-        ] process > NF_FLU:NANOPORE:SOFTWARE_VERSIONS                           -
[-        ] process > NF_FLU:NANOPORE:MULTIQC                                     -
Oops... Pipeline execution stopped with the following message: [WARNING] No absolute output path provided, using current directory as prefix
[INFO] CLAIR3 VERSION: v0.1-r10
[INFO] BAM FILE PATH: /mnt/data6/FLU/work/0e/d71d74763126393c363708386cb7d2/23MB310073.Segment_5_NP.MT467124.bam
[INFO] REFERENCE FILE PATH: /mnt/data6/FLU/work/0e/d71d74763126393c363708386cb7d2/23MB310073.Segment_5_NP.MT467124.reference.fasta
[INFO] MODEL PATH: /mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615
[INFO] OUTPUT FOLDER: /mnt/data6/FLU/work/0e/d71d74763126393c363708386cb7d2/23MB310073.Segment_5_NP.MT467124.clair3
[INFO] PLATFORM: ont
[INFO] THREADS: 2
[INFO] BED FILE PATH: EMPTY
[INFO] VCF FILE PATH: EMPTY
[INFO] CONTIGS: EMPTY
[INFO] CONDA PREFIX: 
[INFO] SAMTOOLS PATH: samtools
[INFO] PYTHON PATH: python3
[INFO] PYPY PATH: pypy3
[INFO] PARALLEL PATH: parallel
[INFO] WHATSHAP PATH: whatshap
[INFO] CHUNK SIZE: 5000000
[INFO] FULL ALIGN PROPORTION: 0.7
[INFO] FULL ALIGN REFERENCE PROPORTION: 0.1
[INFO] PHASING PROPORTION: 0.7
[INFO] ENABLE FILEUP ONLY CALLING: False
[INFO] ENABLE FAST MODE CALLING: True
[INFO] ENABLE CALLING SNP CANDIDATES ONLY: False
[INFO] ENABLE PRINTING REFERENCE CALLS: False
[INFO] ENABLE OUTPUT GVCF: False
[INFO] ENABLE HAPLOID PRECISE MODE: False
[INFO] ENABLE HAPLOID SENSITIVE MODE: True
[INFO] ENABLE INCLUDE ALL CTGS CALLING: True
[INFO] ENABLE NO PHASING FOR FULL ALIGNMENT: False
[INFO] ENABLE REMOVING INTERMEDIATE FILES: False
[INFO] ENABLE PHASING VCF OUTPUT: False
[INFO] ENABLE LONG INDEL CALLING: True

[ERROR] Conda prefix not found, please activate clair3 conda environment first, model path: /mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615
Error executing process > 'NF_FLU:NANOPORE:CLAIR3 (23MB310073|5_NP|MT467124)'

Caused by:
  Missing output file(s) `23MB310073.Segment_5_NP.MT467124.clair3.vcf.gz` expected by process `NF_FLU:NANOPORE:CLAIR3 (23MB310073|5_NP|MT467124)`

Command executed:

  CLAIR_BIN_DIR=$(dirname $(which run_clair3.sh))
  if [[ "/mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615" != "" ]] ; then
      MODEL_PATH=/mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615
  else
      if [[ false = true ]] ; then
          MODEL_PATH="$CLAIR_BIN_DIR/models/r941_prom_sup_g5014"
      else [[ false = false ]]
          MODEL_PATH="/opt/models/r941_prom_sup_g5014"
      fi
  fi
  
  samtools faidx 23MB310073.Segment_5_NP.MT467124.reference.fasta
  
  run_clair3.sh \
      --bam_fn=23MB310073.Segment_5_NP.MT467124.bam \
      --ref_fn=23MB310073.Segment_5_NP.MT467124.reference.fasta \
      --model_path="$MODEL_PATH"\
      --threads=2 \
      --platform="ont" \
      --output=23MB310073.Segment_5_NP.MT467124.clair3 \
      --haploid_sensitive \
      --enable_long_indel \
      --fast_mode \
      --include_all_ctgs
  
  ln -s 23MB310073.Segment_5_NP.MT467124.clair3/merge_output.vcf.gz 23MB310073.Segment_5_NP.MT467124.clair3.vcf.gz
  
  cat <<-END_VERSIONS > versions.yml
  "NF_FLU:NANOPORE:CLAIR3":
      clair3: $(head -n1 23MB310073.Segment_5_NP.MT467124.clair3/run_clair3.log | sed 's/^.*CLAIR3 VERSION: v//; s/ .*$//')
      samtools: $(echo $(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*$//')
  END_VERSIONS

Command exit status:
  0

Command output:
  [WARNING] No absolute output path provided, using current directory as prefix
  [INFO] CLAIR3 VERSION: v0.1-r10
  [INFO] BAM FILE PATH: 23MB310073.Segment_5_NP.MT467124.bam
  [INFO] REFERENCE FILE PATH: 23MB310073.Segment_5_NP.MT467124.reference.fasta
  [INFO] MODEL PATH: /mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615
  [INFO] OUTPUT FOLDER: 23MB310073.Segment_5_NP.MT467124.clair3
  [INFO] PLATFORM: ont
  [INFO] THREADS: 2
  [INFO] BED FILE PATH: EMPTY
  [INFO] VCF FILE PATH: EMPTY
  [INFO] CONTIGS: EMPTY
  [INFO] CONDA PREFIX: 
  [INFO] SAMTOOLS PATH: samtools
  [INFO] PYTHON PATH: python3
  [INFO] PYPY PATH: pypy3
  [INFO] PARALLEL PATH: parallel
  [INFO] WHATSHAP PATH: whatshap
  [INFO] CHUNK SIZE: 5000000
  [INFO] FULL ALIGN PROPORTION: 0.7
  [INFO] FULL ALIGN REFERENCE PROPORTION: 0.1
  [INFO] PHASING PROPORTION: 0.7
  [INFO] ENABLE FILEUP ONLY CALLING: False
  [INFO] ENABLE FAST MODE CALLING: True
  [INFO] ENABLE CALLING SNP CANDIDATES ONLY: False
  [INFO] ENABLE PRINTING REFERENCE CALLS: False
  [INFO] ENABLE OUTPUT GVCF: False
  [INFO] ENABLE HAPLOID PRECISE MODE: False
  [INFO] ENABLE HAPLOID SENSITIVE MODE: True
  [INFO] ENABLE INCLUDE ALL CTGS CALLING: True
  [INFO] ENABLE NO PHASING FOR FULL ALIGNMENT: False
  [INFO] ENABLE REMOVING INTERMEDIATE FILES: False
  [INFO] ENABLE PHASING VCF OUTPUT: False
  [INFO] ENABLE LONG INDEL CALLING: True
  
  [ERROR] Conda prefix not found, please activate clair3 conda environment first, model path: /mnt/data6/FLU/nf-flu/clair3/bin/models/r1041_e82_400bps_sup_g615

Work dir:
  /mnt/data6/FLU/work/0e/d71d74763126393c363708386cb7d2

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

```

May we seek your advice on this issue?

Thank you very much!

Eddie

**Comment by nhhaidee** [comment]
@llk578496 Can you go to work folder `/mnt/data6/FLU/work/0e/d71d74763126393c363708386cb7d2` and attach full log file `${OUPUT_DIR}/run_clair3.log`?

**Comment by llk578496** [comment]
Yes! Please find the file below.
[run_clair3.log](https://github.com/CFIA-NCFAD/nf-flu/files/11592713/run_clair3.log)

**Comment by nhhaidee** [comment]
@llk578496 Can you save model path in local directory instead of mount device? It seems that Docker container does not found model path from mnt device?

**Comment by nhhaidee** [comment]
@peterk87 Can you try with Clair3 latest version v1.0.2? I did try with version v1.0.1 and it worked for me

[INFO] CLAIR3 VERSION: v1.0.1
[INFO] BAM FILE PATH: /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.bam
[INFO] REFERENCE FILE PATH: /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.reference.fasta
[INFO] MODEL PATH: /home/hnguyen/Documents/LabData/Clair3_Model/r1041_e82_400bps_sup_g615
[INFO] OUTPUT FOLDER: /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3
[INFO] PLATFORM: ont
[INFO] THREADS: 2
[INFO] BED FILE PATH: EMPTY
[INFO] VCF FILE PATH: EMPTY
[INFO] CONTIGS: EMPTY
[INFO] CONDA PREFIX:
[INFO] SAMTOOLS PATH: samtools
[INFO] PYTHON PATH: python3
[INFO] PYPY PATH: pypy3
[INFO] PARALLEL PATH: parallel
[INFO] WHATSHAP PATH: whatshap
[INFO] LONGPHASE PATH: EMPTY
[INFO] CHUNK SIZE: 5000000
[INFO] FULL ALIGN PROPORTION: 0.7
[INFO] FULL ALIGN REFERENCE PROPORTION: 0.1
[INFO] PHASING PROPORTION: 0.7
[INFO] MINIMUM MQ: 5
[INFO] MINIMUM COVERAGE: 2
[INFO] SNP AF THRESHOLD: 0.08
[INFO] INDEL AF THRESHOLD: 0.15
[INFO] ENABLE FILEUP ONLY CALLING: False
[INFO] ENABLE FAST MODE CALLING: True
[INFO] ENABLE CALLING SNP CANDIDATES ONLY: False
[INFO] ENABLE PRINTING REFERENCE CALLS: False
[INFO] ENABLE OUTPUT GVCF: False
[INFO] ENABLE HAPLOID PRECISE MODE: False
[INFO] ENABLE HAPLOID SENSITIVE MODE: True
[INFO] ENABLE INCLUDE ALL CTGS CALLING: True
[INFO] ENABLE NO PHASING FOR FULL ALIGNMENT: False
[INFO] ENABLE REMOVING INTERMEDIATE FILES: False
[INFO] ENABLE LONGPHASE FOR INTERMEDIATE VCF PHASING: False
[INFO] ENABLE PHASING FINAL VCF OUTPUT USING WHATSHAP: False
[INFO] ENABLE PHASING FINAL VCF OUTPUT USING LONGPHASE: False
[INFO] ENABLE HAPLOTAGGING FINAL BAM: False
[INFO] ENABLE LONG INDEL CALLING: True
[INFO] ENABLE C_IMPLEMENT: True

+ /opt/bin/scripts/clair3_c_impl.sh --bam_fn /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.bam --ref_fn /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.reference.fasta --threads 2 --model_path /home/hnguyen/Documents/LabData/Clair3_Model/r1041_e82_400bps_sup_g615 --platform ont --output /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3 --bed_fn=EMPTY --vcf_fn=EMPTY --ctg_name=EMPTY --sample_name=SAMPLE --chunk_num=0 --chunk_size=5000000 --samtools=samtools --python=python3 --pypy=pypy3 --parallel=parallel --whatshap=whatshap --qual=2 --var_pct_full=0.7 --ref_pct_full=0.1 --var_pct_phasing=0.7 --snp_min_af=0.08 --indel_min_af=0.15 --min_mq=5 --min_coverage=2 --min_contig_size=0 --pileup_only=False --gvcf=False --fast_mode=True --call_snp_only=False --print_ref_calls=False --haploid_precise=False --haploid_sensitive=True --include_all_ctgs=True --no_phasing_for_fa=False --pileup_model_prefix=pileup --fa_model_prefix=full_alignment --remove_intermediate_dir=False --enable_phasing=False --enable_long_indel=True --keep_iupac_bases=False --use_gpu=False --longphase_for_phasing=False --longphase=EMPTY --use_whatshap_for_intermediate_phasing=True --use_longphase_for_intermediate_phasing=False --use_whatshap_for_final_output_phasing=False --use_longphase_for_final_output_phasing=False --use_whatshap_for_final_output_haplotagging=False

[INFO] Check environment variables
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/log
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp/pileup_output
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp/merge_output
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp/phase_output
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp/gvcf_tmp_output
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp/full_alignment_output
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp/phase_output/phase_vcf
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp/phase_output/phase_bam
[INFO] Create folder /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/tmp/full_alignment_output/candidate_bed
[INFO] --include_all_ctgs enabled
[93m[WARNING] Please enable --no_phasing_for_fa if calling variant in non-diploid organisms [0m
[INFO] Call variant in contigs: WIN-AH-2022-FAV-0227-2_segment7_M
[INFO] Chunk number for each contig: 1
[93m[WARNING] Current maximum contig length 1002 is much smaller than default chunk size 5000000, You may set a smaller chunk size by setting --chunk_size=$ for better parallelism. [0m
[INFO] 1/7 Call variants using pileup model
Calling variants ...
Total processed positions in WIN-AH-2022-FAV-0227-2_segment7_M (chunk 1/1) : 12
Total time elapsed: 4.20 s

real 0m8.346s
user 0m6.437s
sys 0m1.828s

[INFO] 2/7 Select heterozygous SNP variants for Whatshap phasing and haplotagging
[93m[WARNING] Cannot find any 0/1 variant in pileup output using variant quality cut-off proportion: 0.30000000000000004, total heterozygous variants: 0 [0m
[93m[WARNING] Set low variant quality score cut-off to 0.0 [0m
[INFO] Select heterozygous pileup variants exceeding phasing quality cutoff 0
[INFO] Total heterozygous SNP positions selected: WIN-AH-2022-FAV-0227-2_segment7_M: 0

real 0m0.315s
user 0m0.252s
sys 0m0.056s

[INFO] 3/7 Phase VCF file using Whatshap
This is WhatsHap 1.7 running under Python 3.9.0
Working on 1 sample from 1 family

# Resource usage
Maximum memory usage: 0.038 GB
Time spent reading BAM/CRAM:                    0.0 s
Time spent parsing VCF:                         0.0 s
Time spent selecting reads:                     0.0 s
Time spent phasing:                             0.0 s
Time spent writing VCF:                         0.0 s
Time spent finding components:                  0.0 s
Time spent on rest:                             0.0 s
Total elapsed time:                             0.0 s

real 0m0.651s
user 0m0.480s
sys 0m0.131s

[INFO] 5/7 Select candidates for full-alignment calling
[93m[WARNING] Cannot find any low-quality 0/0 reference calls in pileup output using reference quality cut-off proportion: 0.1, total reference calls: 5 [0m
[93m[WARNING] Set low reference quality score cut-off to 0.0 [0m
[INFO] Set variants quality cutoff 14.0
[INFO] Set reference calls quality cutoff 0.0
[INFO] Low quality reference calls to be processed in WIN-AH-2022-FAV-0227-2_segment7_M: 0
[INFO] Low quality variants to be processed in WIN-AH-2022-FAV-0227-2_segment7_M: 3

real 0m0.363s
user 0m0.264s
sys 0m0.077s

[INFO] 6/7 Call low-quality variants using full-alignment model
Calling variants ...
Total processed positions in WIN-AH-2022-FAV-0227-2_segment7_M (chunk 1/1) : 3
Total time elapsed: 1.82 s

real 0m5.152s
user 0m4.120s
sys 0m0.400s

[INFO] 7/7 Merge pileup VCF and full-alignment VCF
[INFO] Pileup variants processed in WIN-AH-2022-FAV-0227-2_segment7_M: 4
[INFO] Full-alignment variants processed in WIN-AH-2022-FAV-0227-2_segment7_M: 3

real 0m0.702s
user 0m0.428s
sys 0m0.057s

[INFO] Finish calling, output file: /home/hnguyen/Documents/LabData/Avian_Influenza/work/47/d01e3791fde274dd851e291a5dc413/WIN_CFIA_AIV_SAMPLE_96.Segment_7_M.WIN-AH-2022-FAV-0227-2_segment7_M.clair3/merge_output.vcf.gz

real 0m17.748s
user 0m13.221s
sys 0m3.328s

**Comment by peterk87** [comment]
> Also, I think we should provide absolute model directory path.

The model likely needs to be specified as a channel input for the process so that Nextflow can handle it appropriately for the executor and profile being used.

```nextflow
process CLAIR3 {
  //...
  input:
  tuple val(sample), val(segment), val(ref_id), path(ref_fasta), path(bam)
  // optional model_path
  path model_path 
```

I'm working on a solution.

**Comment by peterk87** [comment]
The issue should be resolved by #16 in release [3.1.5](https://github.com/CFIA-NCFAD/nf-flu/releases/tag/3.1.5). Clair3 with a user-specified model should work with Docker, Conda and Singularity.

@llk578496 If you want to test nf-flu v3.1.5,  you can `nextflow pull CFIA-NCFAD/nf-flu` to update your local version of the workflow and rerun the same command above (using `-resume` if you'd like to avoid recalculating unchanged process results). Hopefully the new release will work for you! If not, please let us know and we'll try to fix it.

**Comment by llk578496** [comment]
@peterk87 @nhhaidee It works! Thank you so much for you help!
